### ISSUE TITLE
Slonik is 10x slower when used with ncc

### ISSUE THREAD

**Issue by thomasthiebaud** [vercel/ncc#646]
[Slonik](https://github.com/gajus/slonik) is 10x slower when I compiled my code using ncc. I'm not sure where the issue lies, because `pg` does not have the same issue.

Here is a minimal repo to reproduce the issue https://github.com/thomasthiebaud/slonik-perf-issue

**Comment by guybedford** [comment]
@thomasthiebaud source maps can considerably slow down execution time. Did you try compiling with ncc without the source map?

**Comment by thomasthiebaud** [comment]
@guybedford it does not seem that this is the issue. Without source and without minifying, I still have a huge perf difference for Slonik 

```
pg - query: 6.555ms
ncc-pg - query: 12.474ms

slonik - query: 9.418ms
ncc-slonik - query: 606.6ms
```

You can easily reproduce the issue in that repo https://github.com/thomasthiebaud/slonik-perf-issue

**Comment by guybedford** [comment]
@thomasthiebaud do any of the dependencies have native implementations that are only used when a native platform version is available? ncc not managing to inline such native code resulting in a JS fallback might explain the issue?

**Comment by thomasthiebaud** [comment]
yes slonik has an optional dependency on `pg-native` https://github.com/gajus/slonik/blob/6bafb002e2c7fff468b4baeeb7ae57d8c2dd86bd/src/factories/createPool.ts#L48-L58

**Comment by guybedford** [comment]
Right - it would be worth checking if `pg-native` is working via ncc itself directly.

**Comment by thomasthiebaud** [comment]
It does not seem that `pg-native` has an issue with `ncc`

```
pg - query: 23.174ms
ncc-pg - query: 26.919ms

slonik - query: 8.378ms
ncc-slonik - query: 566.233ms

pg-native - query: 2.083ms
ncc-pg-native - query: 2.627ms
```

**Comment by guybedford** [comment]
Is there any other dependency you think it could be? Because this is performance and not an actual breakage, missing native deps / slow optional deps seems the most likely cause.

**Comment by thomasthiebaud** [comment]
It does not seem like there is another native dependency
